pkg/operator/csi/csidrivercontrollerservicecontroller/helpers.go

```diff
@@ -48,6 +49,28 @@ func WithObservedProxyDeploymentHook() dc.DeploymentHookFunc {
 	}
 }
 
+func WithTrustedCADeploymentHook(
+	configMapNamespace string,
+	configMapName string,
+	configMapInformer corev1.ConfigMapInformer,
+) dc.DeploymentHookFunc {
+	return func(_ *opv1.OperatorSpec, deployment *appsv1.Deployment) error {
+		_, err := configMapInformer.Lister().ConfigMaps(configMapNamespace).Get(configMapName)
+		if apierrors.IsNotFound(err) {
+			return nil
+		}
+		if err != nil {
+			return fmt.Errorf("failed to get ConfigMap %s/%s: %v", configMapNamespace, configMapName, err)
+		}
+		containerNamesString := deployment.Annotations["config.openshift.io/inject-trusted-ca"]
```

Comment:
when the content changes, what manages the rolling restart?

Reply:
This controller resyncs when the CM changes (and this function hook is called again).

Reply:
OK, so I hashed the whole CM and added an annotation to the deployment. When the CM content changes, the deployment will be rolled out again.